app/src/main/java/org/openobservatory/ooniprobe/common/MKCollectorResubmitTask.java

```diff
@@ -82,15 +82,7 @@ protected void onPreExecute() {
     protected Void doInBackground(Integer... params) {
         if (params.length != 2)
             throw new IllegalArgumentException("MKCollectorResubmitTask requires 2 nullable params: result_id, measurement_id");
-        Where<Measurement> msmQuery = SQLite.select().from(Measurement.class)
-                .where(Measurement_Table.is_failed.eq(false))
-                // We check on both the report_id and is_uploaded as we
-                // may have some unuploaded measurements which are marked
-                // as is_uploaded = true, but we always know that those with
-                // report_id set to null are not uploaded
-                .and(OperatorGroup.clause()
-                        .or(Measurement_Table.is_uploaded.eq(false))
-                        .or(Measurement_Table.report_id.isNull()));
+        Where<Measurement> msmQuery = Measurement.selectUploadable();
```

Comment:
I know we can further simplify the code below, but it's probably too much of a burder at this time